Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -0,0 +1,197 @@
+import Foundation
+
+/// Represents a MongoDB connection string.
+/// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/
+public struct MongoConnectionString: Codable, LosslessStringConvertible {
+    /// Represents a connection string scheme.
+    public struct Scheme: LosslessStringConvertible, Equatable {
+        /// Indicates that this connection string uses the scheme `mongodb`.
+        public static let mongodb = Scheme(.mongodb)
+
+        /// Indicates that this connection string uses the scheme `mongodb+srv`.
+        public static let srv = Scheme(.srv)
+
+        /// Internal representation of a scheme.
+        private enum _Scheme: String {
+            case mongodb
+            case srv = "mongodb+srv"
+        }
+
+        private let _scheme: _Scheme
+
+        private init(_ value: _Scheme) {
+            self._scheme = value
+        }
+
+        /// LosslessStringConvertible` protocol requirements
+        public init?(_ description: String) {
+            guard let _scheme = _Scheme(rawValue: description) else {
+                return nil
+            }
+            self.init(_scheme)
+        }
+
+        public var description: String { self._scheme.rawValue }
+    }
+
+    /// A struct representing a network address, consisting of a host and port.
```

Comment:
Now that the port is optional and this struct is a bit more broad in what it represents, we should update the docstring accordingly. It could say something along the lines of 
```
A struct representing a host identifier, consisting of a host and an optional port.
In standard connection strings, this describes the address of a mongod or mongos to connect to.
In mongodb+srv connection strings, this describes a DNS name to be queried for SRV and TXT records.
```

Reply:
done